sqlmesh/core/context.py

```diff
@@ -367,19 +369,22 @@ def __init__(
         self.environment_ttl = self.config.environment_ttl
         self.pinned_environments = Environment.sanitize_names(self.config.pinned_environments)
         self.auto_categorize_changes = self.config.plan.auto_categorize_changes
+        self.default_gateway = gateway or self.config.default_gateway_name
```

Comment:
I suggest to not overload the `default_gateway` here. Got me really confused. Default gateway is the one specified in the configuration. `gateway` is the override user provides in eg. CLI `sqlmesh --gateway <gateway>`. We need a better way to name this variable.

Reply:
Some ideas: `selected_gateway`

Reply:
Yes actually that's much better thank you changed all these